### PR TITLE
Create CreateDetectorTable Documentation

### LINKED ISSUE #27360: Create documentation for the CreateDetectorTable algorithm
The algorithm doesn't have an associated rst file, so clicking the help button in the dialog leads to nowhere. One needs to be created.

### PR DESCRIPTION
**Pull request by PhilColebrooke**
**Description of work.**
This PR adds a page to the developer documentation containing information about the CreateDetectorTable algorithm.

**To test:**
Build the development documentation and check there is a new page with the other algorithm docs for CreateDetectorTable.

Fixes #27360 

No release notes because the algorithm doesn't exist in a release yet.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/algorithms/CreateDetectorTable-v1.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+   ws_detectors = CreateDetectorTable(ws)
+
+   # Print the first row of the table
+   print(ws_detectors.row(0))
+
+Output:
+
+.. testoutput:: CreateDetectorTableExample
+
+  {'Index': 0.0, 'Phi': 0.0, 'Monitor': 'no', 'Spectrum No': 1, 'R': 5.0, 'Detector ID(s)': '100', 'Theta': 0.0}
```
Comment (Harrietbrown) [inline docs/source/algorithms/CreateDetectorTable-v1.rst]:
```suggestion
  {'Index': 0.0, 'Spectrum No': 1, 'Detector ID(s)': '100', 'R': 5.0, 'Theta': 0.0, 'Phi': 0.0, 'Monitor': 'no'}
```
This should fix the failing doctest

**Review by Harrietbrown — CHANGES_REQUESTED** [review]

File: docs/source/algorithms/CreateDetectorTable-v1.rst (hunk 1 above)
Comment (PhilColebrooke) [reply docs/source/algorithms/CreateDetectorTable-v1.rst]: Oh I literally just pushed a change for this. I thought the failure was due to dictionaries being unordered so the output order varied. I've changed the way the output is created so the order is fixed.

Comment (Harrietbrown) [reply docs/source/algorithms/CreateDetectorTable-v1.rst]: Your solution is better. this was a quick fix but much more fragile than yours.

**Review by Harrietbrown — APPROVED** [review]
Docs compile correctly. good job.

**Pull request merged**